### ISSUE TITLE
InputBag::get() method incorrectly reports type error for bool default parameter

### ISSUE THREAD

**Issue by sfmok** [phpstan/phpstan-symfony#453]
PHPStan is incorrectly reporting a type error when using `false` as the default parameter for `Symfony\Component\HttpFoundation\InputBag::get()` method. 

The error states that the method expects `string|null` but `false` is given, which is incorrect according to the actual method signature.

PHPStan reports the following error:
```
Parameter #2 $default of method Symfony\Component\HttpFoundation\InputBag<string>::get() expects string|null, false given.
```

Code Example

```php
use Symfony\Component\HttpFoundation\Request;

$request = new Request();
$isPreview = $request->query->get('preview', false); // This triggers the error
```

Environment Details
- PHPStan Version: 1.12.25
- PHPStan Symfony Extension: 1.4.15
- Symfony Version: 6.4.*
- PHP Version: 8.2.20

The actual `InputBag::get()` method signature is:
```php
/**
 * Returns a scalar input value by name.
 *
 * @param string|int|float|bool|null $default The default value if the input key does not exist
 */
public function get(string $key, mixed $default = null): string|int|float|bool|null
{
    if (null !== $default && !\is_scalar($default) && !$default instanceof \Stringable) {
        throw new \InvalidArgumentException(sprintf('Expected a scalar value as a 2nd argument to "%s()", "%s" given.', __METHOD__, get_debug_type($default)));
    }

    $value = parent::get($key, $this);

    if (null !== $value && $this !== $value && !\is_scalar($value) && !$value instanceof \Stringable) {
        throw new BadRequestException(sprintf('Input value "%s" contains a non-scalar value.', $key));
    }

    return $this === $value ? $default : $value;
}
```

This appears to be an issue with PHPStan's Symfony extension having an outdated or incorrect type definition for the InputBag::get() method. The extension seems to be using an older type definition that doesn't match the current Symfony implementation.

**Comment by ben-at-axiom** [comment]
I'm having a similar issue, I believe. for a method with this signature:

```
public static function createFromRequest(InputBag $input): SomeDto
```

phpstan complains:

```
Method ... has parameter $input with generic class Symfony\Component\HttpFoundation\InputBag but does not specify its types: TValue
```

however `InputBag` is not a generic class

**Comment by ondrejmirtes** [comment]
It is https://github.com/phpstan/phpstan-symfony/blob/2.0.x/stubs/Symfony/Component/HttpFoundation/InputBag.stub

**Comment by ben-at-axiom** [comment]
yeah, that's an inappropriate definition of `InputBag`. a single instance of an `InputBag` can return many different values per `->get(...)` call. templating the method makes sense (i.e. `$inputBag->get<string>('someParameter');`); templating the entire class doesn't.

**Comment by ondrejmirtes** [comment]
You can just type `@param InputBag<string|int|float|bool> $bag` if it can contain anything in that place.
